src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -19,6 +19,8 @@ public partial class Ping
         private const int IcmpHeaderLengthInBytes = 8;
         private const int MinIpHeaderLengthInBytes = 20;
         private const int MaxIpHeaderLengthInBytes = 60;
+        private static bool _sendIpHeader = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+        private static bool _needsConnect = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
```

Comment:
These are very fast calls; I'm not sure they need to be cached like this.  But if you think it's important, make 'em readonly as suggested.